returnn/training.py

```diff
@@ -177,10 +183,25 @@ def __init__(
             "time": time_rqmt,
         }
 
-        if self.use_horovod:
-            self.rqmt["cpu"] *= self.horovod_num_processes
-            self.rqmt["gpu"] *= self.horovod_num_processes
-            self.rqmt["mem"] *= self.horovod_num_processes
+        if self.multi_node_slots:
+            assert (
+                self.horovod_num_processes
+            ), "multi_node_slots only supported together with Horovod currently"
+            assert self.horovod_num_processes >= self.multi_node_slots
+            assert self.horovod_num_processes % self.multi_node_slots == 0
+            self.rqmt["multi_node_slots"] = self.multi_node_slots
+
+        if (self.horovod_num_processes or 1) > (self.multi_node_slots or 1):
+            assert self.horovod_num_processes % (self.multi_node_slots or 1) == 0
+            self.rqmt["cpu"] *= self.horovod_num_processes // (
+                self.multi_node_slots or 1
+            )
+            self.rqmt["gpu"] *= self.horovod_num_processes // (
```

Comment:
Is this expect to have non-fitting ratios? So what should happen if `self.horovod_num_processes` is 3 but `self.multi_node_slots` is 2? Does that even make sense? I assume you can not get 2 gpus on node and 1 on the other. And in this case we now get 1 GPU requested per node right?

Reply:
There is an assert for it above.

Reply:
ah did not see that

Reply:
>I assume you can not get 2 gpus on node and 1 on the other. 

Technically, when you use horovod_num_processes==multi_node_slots==3 then two slots can be on the same node. But each would get the indicated rqmt individually.

Reply:
Note that we don't really have any specific logic at all for this. We just pass on these numbers to SGE `-pe` and `mpirun` respectively. I assume there are countless of possible configurations on how exactly it would then do the distribution.